### PR TITLE
(#166) finance reports

### PR DESCRIPTION
**Pull request by rathorevaibhav**
#166 

Needs to be reviewed and merged after #177

### CODE REVIEW

File: app/Http/Controllers/Finance/ReportsController.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            return view('finance.reports.index')->with([
+                'invoices' => $invoices,
+                'report' => self::getReportDetails($invoices),
+                'startDate' => $startDate,
+                'endDate' => $endDate,
+            ]);
+        }
+
+        $invoices = Invoice::all();
+        return view('finance.reports.index')->with([
```
Comment (nishanth-kd) [inline app/Http/Controllers/Finance/ReportsController.php]: A similar feedback here. I would suggest you look at this again. I see this code can be optimised in terms of number of lines.

File: app/Http/Controllers/Finance/ReportsController.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     * @return array
+     */
+    public static function getReportDetails($invoices)
+    {
+        $report = [];
+        $report['tds'] = 0;
+        $report['gst'] = 0;
+        foreach ($invoices as $invoice) {
+            $report['gst'] += $invoice->gst;
+            if (!isset($report['sentAmount'][$invoice->currency_sent_amount])) {
```
Comment (nishanth-kd) [inline app/Http/Controllers/Finance/ReportsController.php]: Can these Calculations be done via MYSQL instead by using MYSQLs Caculation functions to aggregate values in particular columns. 

I think you should give it a shot :)

File: app/Http/Controllers/Finance/ReportsController.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        $report['tds'] = 0;
+        $report['gst'] = 0;
+        foreach (config('constants.currency') as $currency => $currencyMeta) {
+            $report['sentAmount'][$currency] = 0;
+            $report['paidAmount'][$currency] = 0;
+            $report['transactionCharge'][$currency] = 0;
+            $report['transactionTax'][$currency] = 0;
+        }
+        foreach ($invoices as $invoice) {
+            $report['gst'] += $invoice->gst;
```
Comment (nishanth-kd) [inline app/Http/Controllers/Finance/ReportsController.php]: couldn't sum up the GST via SQL ?

Comment (nishanth-kd) [reply app/Http/Controllers/Finance/ReportsController.php]: It appears my previous comments haven't been address. Please don't assign for review in future unless previous feedbacks haven't been addressed.

Comment (rathorevaibhav) [reply app/Http/Controllers/Finance/ReportsController.php]: @nishanth-kd I had read your previous comment on getting report calculations from the SQL but missed addressing.

Right now, we're not very sure what exactly we need in reporting and how exactly that'll be used. Considering that and also we have to make these calculations in SQL, the code is wrapped within a single method which we can remove later when we move to SQL.

I've created a tracking for this #182 . Once we test and use the reports feature, I think then it'll be better to move to SQL for further performance enhancements.

**Pull request merged**